Feat: Add role descriptions to select inputs

Mikehrn opened the pull request:
<img width="544" alt="Screenshot 2024-10-09 at 16 17 31" src="https://github.com/user-attachments/assets/7ce45f0b-d644-414b-93ef-8b0c8dc6755e">

<img width="558" alt="Screenshot 2024-10-09 at 16 12 47" src="https://github.com/user-attachments/assets/f3e031db-fa03-445f-8a7b-c8e36e8257a0">

<img width="939" alt="Screenshot 2024-10-09 at 16 07 57" src="https://github.com/user-attachments/assets/e4fdcb3e-0ef6-4f96-8750-e3642d74c4ae">

benjaminvo reviewed (changes requested):
A couple of things:

1) There's a pretty aggressive max height on these. Would be great to increase it a bit so we don't get the scrolling inside. Also, there's no checkmark for the selected item in this invite modal?
![CleanShot 2024-10-09 at 21 52 35@2x](https://github.com/user-attachments/assets/99bb8e59-4df8-4045-b814-12daa6727eea)

2) I see this in server settings:
![CleanShot 2024-10-09 at 21 55 44@2x](https://github.com/user-attachments/assets/78c62dd3-97a1-4de6-ab01-046cab3af095)

3) Please remove the hover effect from the selected item. I personally don't like when it switches from a darker gray to a lighter gray on hover :).

4) Minor: Let's do `pr-8` instead of `pr-9` for the select items. It wraps the copy a bit better in some of the select inputs (I don't want to change the copy 😬).

Mikehrn commented:
Updated the styling and fixed the role display issue

<img width="636" alt="Screenshot 2024-10-09 at 22 24 53" src="https://github.com/user-attachments/assets/6ba9e0b3-6752-4329-805a-9b279c2551df">

<img width="922" alt="Screenshot 2024-10-09 at 22 36 39" src="https://github.com/user-attachments/assets/d1a32924-929f-4cb6-a37b-43a92cca7d62">

<img width="557" alt="Screenshot 2024-10-09 at 22 36 32" src="https://github.com/user-attachments/assets/c31b650d-ce6e-4503-bcf3-0f3eb863d948">

benjaminvo commented:
Works very nicely now! Minor thing we could do now that we're touching these:

Add a hover tooltip to the Guest server role when it's disabled saying "The Guest user role isn't enabled on the server". And the tooltip direction should probably be to the right or left so it doesn't block the text above or is positioned outside the viewport below.

![CleanShot 2024-10-09 at 22 47 36@2x](https://github.com/user-attachments/assets/34bd4c51-58a2-4dbe-aeaf-acebe8bdb1d9)

Something similar could be done for when the Owner item is disabled when inviting to a project but I guess this will change. But here it could say "Guests cannot be project owners".

![CleanShot 2024-10-09 at 22 45 44@2x](https://github.com/user-attachments/assets/e1edb75d-8aec-4eab-9321-4a32b000c5bb)
